### PR TITLE
data/selinux: allow mandb_t to search /var/lib/snapd

### PR DESCRIPTION
**Pull request by bboozzoo**
Mandb tends to look everywhere in the filesystem in search for directories
containing manpages. Since cache update is executed under a targeted policy,
with mandb_t type, attempts to poke under /var/lib/snapd (snappy_var_lib_t)
raise SELinux denials like this:

```
time->Tue Aug 13 11:44:03 2019
type=AVC msg=audit(1565696643.557:298): avc:  denied  { search } for  pid=22851
         comm="mandb" name="snapd" dev="vda1" ino=524751
         scontext=system_u:system_r:mandb_t:s0
         tcontext=system_u:object_r:snappy_var_lib_t:s0
         tclass=dir permissive=0
```

Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1648701

cc @Conan-Kudo

### CODE REVIEW

File: data/selinux/snappy.te [hunk 1]
```diff
@@ -654,3 +654,11 @@ allow init_t snappy_cli_t:unix_stream_socket create_stream_socket_perms;
 # declare fs_type(snappy_snap_t) outside of core policy, add explicit permission
 # instead
 allow init_t snappy_snap_t:filesystem remount;
+
+########################################
+#
+# extra policy for mandb_t
+#
+gen_require(` type mandb_t; ')
```
Comment (stolowski) [inline data/selinux/snappy.te]: Super nitpick^2: we seem to be using newlines in gen_require, e.g.
```
gen_require(`
   type mandb_t;
')
```

**Comment by Conan-Kudo** [comment]
@mvo5 @bboozzoo Please cherry pick into 2.40.

File: data/selinux/snappy.te [hunk 2]
```diff
@@ -633,7 +663,9 @@ allow init_t snappy_unconfined_snap_t:process { sigkill signull signal };
 
 # snap tools can be invoked by the regular user, make sure that things get
 # proper labels
-gen_require(` type unconfined_t; ')
```
Comment (zyga) [inline data/selinux/snappy.te]: I find all the changes above noisy, there seem unrelated to the actual change below, am I correct?

File: data/selinux/snappy.te [hunk 3]
```diff
@@ -654,3 +686,13 @@ allow init_t snappy_cli_t:unix_stream_socket create_stream_socket_perms;
 # declare fs_type(snappy_snap_t) outside of core policy, add explicit permission
 # instead
 allow init_t snappy_snap_t:filesystem remount;
+
+########################################
+#
+# extra policy for mandb_t
+#
+gen_require(`
```
Comment (zyga) [inline data/selinux/snappy.te]: +1 on this part

File: data/selinux/snappy.te (hunk 2 above)
Comment (bboozzoo) [reply data/selinux/snappy.te]: It's in a separate patch though. FWIW, we should have used the format from the start

**Comment by mvo5** [comment]
Thanks! I cherry picked the two commits to 2.40
